From 088e66778dc9cb8ca17889acfce0004e399bf029 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:43 -0400
Subject: [ethernet] mlx4: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <2c34ad2a7384bda8baa64e0abcc00a81c21d0f92.1410425016.git.avadai@redhat.com>
Patchwork-id: 93609
O-Subject: [Patch RHEL7.1 V1 047/197] mlx4: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range() and pci_enable_msix_range()
interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: "David S. Miller" <davem@davemloft.net>
Cc: Amir Vadai <amirv@mellanox.com>
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Acked-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 66e2f9c1de5550ab8e86ec6d33bf3f4215d389b8)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/main.c b/drivers/net/ethernet/mellanox/mlx4/main.c
index 0ba02aa..3da6adc 100644
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -1983,7 +1983,6 @@ static void mlx4_enable_msi_x(struct mlx4_dev *dev)
  int nreq = min_t(int, dev->caps.num_ports *
     min_t(int, netif_get_num_default_rss_queues() + 1,
           MAX_MSIX_P_PORT) + MSIX_LEGACY_SZ, MAX_MSIX);
- int err;
  int i;
 
  if (msi_x) {
@@ -1997,23 +1996,13 @@ static void mlx4_enable_msi_x(struct mlx4_dev *dev)
   for (i = 0; i < nreq; ++i)
    entries[i].entry = i;
 
- retry:
-  err = pci_enable_msix(dev->pdev, entries, nreq);
-  if (err) {
-   /* Try again if at least 2 vectors are available */
-   if (err > 1) {
-    mlx4_info(dev, "Requested %d vectors, "
-       "but only %d MSI-X vectors available, "
-       "trying again\n", nreq, err);
-    nreq = err;
-    goto retry;
-   }
+  nreq = pci_enable_msix_range(dev->pdev, entries, 2, nreq);
+
+  if (nreq < 0) {
    kfree(entries);
    goto no_msi;
-  }
-
-  if (nreq <
-      MSIX_LEGACY_SZ + dev->caps.num_ports * MIN_MSIX_P_PORT) {
+  } else if (nreq < MSIX_LEGACY_SZ +
+      dev->caps.num_ports * MIN_MSIX_P_PORT) {
    /*Working in legacy mode , all EQ's shared*/
    dev->caps.comp_pool           = 0;
    dev->caps.num_comp_vectors = nreq - 1;
-- 
1.7.1